src/acl/CertificateData.cc

```diff
@@ -107,14 +105,10 @@ ACLCertificateData::parse()
                 return;
             }
 
-            /* an acl must use consistent attributes in all config lines */
-            if (attribute) {
-                if (strcasecmp(newAttribute, attribute) != 0) {
-                    debugs(28, DBG_CRITICAL, "FATAL: An acl must use consistent attributes in all config lines (" << newAttribute << "!=" << attribute << ").");
-                    self_destruct();
-                    return;
-                }
-            } else {
```

Comment:
AFAICT, the above deleted lines represent the reduction in code duplication -- the primary focus of this PR. It is not much, but still worth doing IMO. The rest is an "upgrade" of `attribute` to SBuf (to become compatible with SetAclKey()), moving strtokFile() outside SetAclKey() (because the values.insert() code above needs to handle tokens that are ACL values rather than optional ACL attributes), and associated code synchronization changes.

I am aware that `-sha1` processing is currently broken, but this PR manages to avoid touching the buggy code, and will not make fixing that bug more difficult AFAICT. That fix will need a dedicated PR.